docs/src/architecture/08_concepts/cardano_chain_follower/follower_instances.md

```diff
@@ -0,0 +1,122 @@
+    the moment they become equal one of the streams is closed and all block data is now taken from a single stream.
+    * This is likely to occur, for example a follower from Tip and Genesis.
+    Once the Genesis follower has reached tip we now have two followers from Tip when only 1 is required.
+4. An ideal solution may need multiple techniques.
+
+*Note: There may be other techniques and these are just some obvious solutions to the issue.*
+
+For the initial implementation of the Follower, we will employ technique 2 and a limited version of Technique 3 ONLY.
+
+#### Delaying starting of the streams
+
+When a stream is subscribed, and it results in a new stream, then the connection to that relay node will not happen for 5 seconds.
+Any further subscriptions to the same stream can then be added to the pending connection.
+Once the connection is made, then further subscriptions will only be added to it if they are valid for that connection.
+Otherwise a second connection is established, again with its own 5 second start delay.
+
+This delay occurs regardless of where the stream is to start from Tip, Genesis or a particular block.
+
+#### Optimizing the streams
+
+If we have two followers for a particular network, and they are BOTH on TIP,
+then one is stopped and all subscribers for it are moved to the other.
+
+This needs to check that not only are they both reported to be on TIP, but that the TIPs are identical.
+
+We do not need to do any other stream optimization at this time.
+
+The purpose of this is to ensure that once followers have fully synced we are not running excessive followers
+that are doing the same work and reporting the same events.
+
+### Handling "Continue" connections
+
+There is an option to listen to a stream from "Continue" but what does "Continue" actually mean?
+If there are multiple streams it could match on, which one does it match?
+
+We will use the rule that if "Continue" is specified, it means "Follow the blockchain with the earliest blocks".
+
+So if there are three followers, one from Genesis, one from block 10,000 and one from Tip.  
+"Continue" will join the Genesis follower.
+
+There is a further condition for "Continue" which is what happens if "Continue" is specified but there are no
+blockchain followers running or pending for the same Application?
+This can only happen either at the start of world for a particular Application OR
+an Application has unsubscribed from all blockchain events.
+
+In this case the "Continue" subscription should be added to list of continue subscriptions for each application.
+When a block is received and about to be sent as an event to an application, we can check the "Continue" list for that app,
+
+If there are subscriptions in that list,
+and the event is from the oldest active follower for the app then they are removed from that list,
+and added as active subscriptions to the selected follower.
```

Comment:
I didn't get this because if none of the app's modules are subscribed yet and one of them wants to "continue", should the runtime extension find (or spawn) a follower to follow from either the genesis (if it's an unitialized app) or the point where it previously stopped/unsubscribed?

Reply:
We don't know what order modules will start in,  so its possible that one module is designed to follow the stream of another.  I am thinking that "follow" might be not sufficient after thinking this through.

For now I think we should do the following, because I am not 100% sure `continue` as its currently cast is workable.
Just comment out `continue`, we never said it would be an option for M2.  That simplifies things a lot.
We can then get the follower working in a naive way very easily, we just need to track what modules are following what, and we can have a unique follower for each possibility in our first iteration.
Clearly its not optimal, but it would work ok.

Reply:
This document can stand as is, because its a design document.  We don't need to implement all of it in your current work, and we can revise it as needed.